vowpalwabbit/io_buf.cc

```diff
@@ -158,3 +158,26 @@ void io_buf::close_file_or_socket(int f)
   close(f);
 #endif
 }
+
+//make all data that was writen available for reading
+void io_buf::flip_from_write_to_read()
+{
+  space.end() = head;
+  head = space.begin();
+}
+
+size_t io_buf::copy_to(void *dst, size_t max_size)
+{
+  size_t to_copy = std::min(available_to_read(), max_size);
+  memcpy(dst, space.begin(), to_copy);
```

Comment:
What is this copying? The data thats already been read?

Reply:
Looks like it is copying out of the internal buffer